### PR TITLE
feat: Select darkmode in app.cal.com

### LINKED ISSUE #11997: No separate option to enable dark mode
### Is your proposal related to a problem?
Yes
  

(Write your answer here.)
So almost every website or saas these days have a separate button within there UI to enable dark mode it's a very small but big thing that effects usability for the service because let's say I only want dark mode in notion or cal.com but not on twitter or Instagram. By default it would force dark mode if I changed my device theme to dark and all other websites where I don't want to see dark mode but without doing it cal.com wouldn't change to dark mode which is a problem and I think it should fixed it's a small thing but does effect usability. 
### Describe the solution you'd like
Add a separate option to change cal.com theme to dark mode in settings or main UI without changing my device theme. 

(Describe your proposed solution here.)
Written above
### Describe alternatives you've considered
There's no alternative whatsoever I have to change my device theme to dark then only it works no other way. 

(Write your answer here.)

### Additional context

(Write your answer here.)

### Requirement/Document

(Share it here.)

### PR DESCRIPTION
**Pull request by swh00tw**
## What does this PR do?

Add a section in the appearance page to switch dark/light mode for the Main page(not the booking page).

Solved #11997

https://github.com/calcom/cal.com/assets/71842426/6c165617-283e-4aee-9845-8b3e660985d2

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is a UI/UX design document, please, share it here.

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

1. Go to `localhost:3000`
2. Go to your setting page's appearance tab
3.  Switch the "App theme", and you will see the color change of the UI. (Cal.com's system default is dark mode)

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

## Note

Currently, for the wording here, it's only in English for now (I've asked a person from Cal.com in the OSS hackathon, and he said using English for now is good). We need to create another PR to support all other languages.

Screenshot 2023-12-02 at 12 12 24 PM

### CODE REVIEW

File: apps/web/pages/settings/my-account/appearance.tsx [hunk 1]
```diff
@@ -137,6 +151,56 @@ const AppearanceView = ({
   return (
     <div>
       <Meta title={t("appearance")} description={t("appearance_description")} borderInShellHeader={false} />
+      <div className="border-subtle mt-6 flex items-center rounded-t-lg border p-6 text-sm">
+        <div>
+          {/* TODO: support i18n: currently it's hard code now, we should support other languages in other PR */}
```
Comment (Udit-takkar) [inline apps/web/pages/settings/my-account/appearance.tsx:156]: you have to just add translation for english language in apps/web/public/static/locales/en/common.json. we'll add translations for other languages later.

**Comment by Udit-takkar** [comment]
@swh00tw there are some type error and make sure to sign CLA https://github.com/calcom/cal.com/pull/12633#issuecomment-1837208236

**Comment by swh00tw** [comment]
> @swh00tw there are some type error and make sure to sign CLA [#12633 (comment)](https://github.com/calcom/cal.com/pull/12633#issuecomment-1837208236)

Okay! I am working on getting rid of that now.

File: apps/web/pages/settings/my-account/appearance.tsx (hunk 1 above)
Comment (Udit-takkar) [reply apps/web/pages/settings/my-account/appearance.tsx:156]: For example:- If you have to use text "Reason for reschedule".
add a key value pair in en/common.json "reason_for_reschedule" :  "Reason for reschedule". 

and use it here t("reason_for_reschedule") like you can see other parts of code

Comment (swh00tw) [reply apps/web/pages/settings/my-account/appearance.tsx:156]: Gotcha, I think I did that. I added a new key called "app_theme". However, the vscode extension kept saying something like "ERROR: "app_theme" is not found in source with language tag 'en'". And the website isn't reflecting the new wording as well. Still trying to figure out why.

Screenshot 2023-12-02 at 12 43 01 PM

Screenshot 2023-12-02 at 12 44 30 PM

Comment (swh00tw) [reply apps/web/pages/settings/my-account/appearance.tsx:156]: After I `yarn dx`. The website has reflected the changes. Nice. But the vscode extension warning is still there

Comment (Udit-takkar) [reply apps/web/pages/settings/my-account/appearance.tsx:156]: @swh00tw no issue. this happens sometimes. just make sure you have added all the keys to common.json that you are using even though you see this error in extension. it gets fixed after some time

**Comment by swh00tw** [comment]
> @swh00tw there are some type errors and make sure to sign CLA [#12633 (comment)](https://github.com/calcom/cal.com/pull/12633#issuecomment-1837208236)

Solved the typing error. Notify me if there is any problem :)

File: apps/web/pages/settings/my-account/appearance.tsx [hunk 2]
```diff
@@ -137,6 +151,59 @@ const AppearanceView = ({
   return (
     <div>
       <Meta title={t("appearance")} description={t("appearance_description")} borderInShellHeader={false} />
+      <div className="border-subtle mt-6 flex items-center rounded-t-lg border p-6 text-sm">
+        <div>
+          {/* TODO: support i18n: we should support other languages in other PR */}
```
Comment (Udit-takkar) [inline apps/web/pages/settings/my-account/appearance.tsx:156]: NIT: remove this

File: apps/web/pages/settings/my-account/appearance.tsx [hunk 3]
```diff
@@ -137,6 +151,59 @@ const AppearanceView = ({
...
+          <p className="text-default text-base font-semibold">{t("app_theme")}</p>
+          <p className="text-default">{t("app_theme_applies_note")}</p>
+        </div>
+      </div>
+      <Form
+        form={userAppThemeFormMethods}
+        handleSubmit={(values) => {
+          mutation.mutate({
+            // Radio values don't support null as values, therefore we convert an empty string
+            // back to null here.
```
Comment (Udit-takkar) [inline apps/web/pages/settings/my-account/appearance.tsx:166]: NIT: you can remove this comment

File: apps/web/pages/settings/my-account/appearance.tsx [hunk 4]
```diff
@@ -137,6 +151,59 @@ const AppearanceView = ({
   return (
     <div>
       <Meta title={t("appearance")} description={t("appearance_description")} borderInShellHeader={false} />
+      <div className="border-subtle mt-6 flex items-center rounded-t-lg border p-6 text-sm">
+        <div>
+          {/* TODO: support i18n: we should support other languages in other PR */}
+          <p className="text-default text-base font-semibold">{t("app_theme")}</p>
+          <p className="text-default">{t("app_theme_applies_note")}</p>
```
Comment (Udit-takkar) [inline apps/web/pages/settings/my-account/appearance.tsx:158]: we should use the text mentioned in the issue comment by peer

https://github.com/calcom/cal.com/issues/11997#issuecomment-1828153608

"Dashboard theme"

**Review by Udit-takkar — COMMENTED** [review]
@swh00tw Nice attempt but there are some issues

When I tried to change the app theme then https://www.loom.com/share/98f941133c5a46fd9c7b86f6ce10ca97?sid=b178f310-0963-48f9-856d-414996203b53

there were some issues with colors and i had to reload to make sure correct colors are loaded

File: apps/web/pages/settings/my-account/appearance.tsx (hunk 2 above)
Comment (swh00tw) [reply apps/web/pages/settings/my-account/appearance.tsx]: Removed

File: apps/web/pages/settings/my-account/appearance.tsx (hunk 3 above)
Comment (swh00tw) [reply apps/web/pages/settings/my-account/appearance.tsx]: Removed

File: apps/web/pages/settings/my-account/appearance.tsx (hunk 4 above)
Comment (swh00tw) [reply apps/web/pages/settings/my-account/appearance.tsx:157]: Got it! I've modified the text. I should read the issue more carefully next time. Thanks!

**Comment by Udit-takkar** [comment]
Not only the toast component but even buttons, switches .. etc.

**Comment by swh00tw** [comment]
I see. I can reproduce this error. Do you have any idea why this happened or any suggested way to resolve this?
I am suspecting that maybe `useTheme` miss something so that the styles of the components aren't not changed

**Comment by Udit-takkar** [comment]
@sean-brydon might know about this

**Review by Udit-takkar — COMMENTED** [review]
If i select 'Light mode' in Dashboard Theme and 'Dark Mode' in Booking Page Theme and go the booking page then i see the booking page in dark mode but the dashboard also changes to dark mode. can we fix this?

https://github.com/calcom/cal.com/assets/53316345/334e8722-b0c9-434b-b1c2-375e6b92f5a2

**Comment by Udit-takkar** [comment]
cc: @sean-brydon for any feedback

**Comment by PeerRich** [comment]
@swh00tw can you take a look at [this](https://github.com/calcom/cal.com/pull/12633#pullrequestreview-1838258444)?

File: apps/web/pages/event-types/index.tsx [hunk 5]
```diff
@@ -930,6 +931,7 @@ const EventTypesPage = () => {
   const orgBranding = useOrgBranding();
   const routerQuery = useRouterQuery();
   const filters = getTeamsFiltersFromQuery(routerQuery);
+  useTheme(user?.appTheme);
```
Comment (sean-brydon) [inline apps/web/pages/event-types/index.tsx:934]: Do we only want to do this in `event-types/index` this should probably be handled in `Shell:356` - This is where we also handle settings brandColours for a user

**Review by sean-brydon — CHANGES_REQUESTED** [review]
Overall this looks pretty solid outside of the feedback @Udit-takkar has mentioned. 

If these changes are not addressed we will take over this PR in a few days.

**Comment by swh00tw** [comment]
> Overall this looks pretty solid outside of the feedback @Udit-takkar has mentioned.
> 
> If these changes are not addressed we will take over this PR in a few days.

Hey guys, really sorry I missed the previous comments. I am kinda busy these days and have little time working on this. I think the best option for this project is to let you guys take over this PR.

**Comment by sean-brydon** [comment]
> > Overall this looks pretty solid outside of the feedback @Udit-takkar has mentioned.
> > If these changes are not addressed we will take over this PR in a few days.
> 
> Hey guys, really sorry I missed the previous comments. I am kinda busy these days and have little time working on this. I think the best option for this project is to let you guys take over this PR.

Awesome - thanks for your response. I will take over this PR sometime next week when i get time. Thanks for your contribution.

**Comment by keithwillcode** [comment]
Closing in favor of https://github.com/calcom/cal.com/pull/13779

**Pull request closed without merging**